File: src/main/java/com/dyonovan/jatm/common/container/machine/ContainerElectricFurnace.java
package com.dyonovan.jatm.common.container.machine;

import com.dyonovan.jatm.common.container.BaseContainer;
import com.dyonovan.jatm.common.tileentity.machine.TileElectricFurnace;
import net.minecraft.entity.player.InventoryPlayer;
import net.minecraft.inventory.Slot;
import net.minecraft.inventory.SlotFurnaceOutput;

public class ContainerElectricFurnace extends BaseContainer {

    TileElectricFurnace tile;

    public ContainerElectricFurnace(InventoryPlayer inventory, TileElectricFurnace tileEntity) {
        this.tile = tileEntity;

        addSlotToContainer(new Slot(tileEntity, TileElectricFurnace.INPUT_SLOT, 57, 34));
        addSlotToContainer(new SlotFurnaceOutput(inventory.player, tileEntity, TileElectricFurnace.OUTPUT_SLOT, 117, 34));
        bindPlayerInventory(inventory, 8, 84);
    }
}

File: src/main/scala/com/dyonovan/neotech/common/container/machines/processors/ContainerElectricFurnace.scala
package com.dyonovan.neotech.common.container.machines.processors

import com.dyonovan.neotech.common.container.machines.ContainerAbstractMachine
import com.dyonovan.neotech.common.container.slot.SlotFurnaceOutputItemHandler
import com.dyonovan.neotech.common.tiles.machines.processors.TileElectricFurnace
import net.minecraft.entity.player.InventoryPlayer

class ContainerElectricFurnace(playerInventory: InventoryPlayer, tile: TileElectricFurnace) extends
        ContainerAbstractMachine(playerInventory, tile) {

    addSlotToContainer(new RestrictedSlot(tile, 0, 56, 35))
    addSlotToContainer(new SlotFurnaceOutputItemHandler(playerInventory.player, tile, 1, 116, 35))
    addPlayerInventorySlots(8, 84)
}
